Library/Homebrew/dev-cmd/audit.rb

```diff
@@ -925,16 +927,17 @@ def audit_revision_and_version_scheme
           newest_committed_version ||= previous_version
           newest_committed_checksum ||= previous_checksum
           newest_committed_revision ||= previous_revision
+          newest_committed_url ||= stable.url
         end
 
         break if previous_version && current_version != previous_version
         break if previous_revision && current_revision != previous_revision
       end
 
-      if current_version == newest_committed_version &&
+      if current_url == newest_committed_url &&
```

Comment:
```suggestion
      if current_version == newest_committed_version &&
         current_url == newest_committed_url &&
```

Are there ever situations where we have URLs like:

```ruby
  url "https://github.com/vim-volt/volt.git",
    tag:      "v0.3.7",
    revision: "e604467d8b440c89793b2e113cd241915e431bf9"
```

But that _also_ have a `sha256`?

These types of URLs won't change on version bumps so it might be good to keep the version check as well. However, I'm not sure off the top of my head whether any of these types of URLs would have a `sha256`. If not, it may not matter.

Reply:
I think there's an audit to check having a tag-and-revision style `url` and also a `sha256`.

Reply:
There aren't many, but I found a few examples that I think would be problematic:
- [`chcase`](https://github.com/Homebrew/homebrew-core/blob/master/Formula/chcase.rb)
- [`google-sql-tool`](https://github.com/Homebrew/homebrew-core/blob/master/Formula/google-sql-tool.rb)
- [`lockrun`](https://github.com/Homebrew/homebrew-core/blob/master/Formula/lockrun.rb)
- [`magnetix`](https://github.com/Homebrew/homebrew-core/blob/master/Formula/magnetix.rb)
- [`malbolge`](https://github.com/Homebrew/homebrew-core/blob/master/Formula/malbolge.rb)
- [`pidof`](https://github.com/Homebrew/homebrew-core/blob/master/Formula/pidof.rb)
- [`plod`](https://github.com/Homebrew/homebrew-core/blob/master/Formula/plod.rb)
- [`rinetd`](https://github.com/Homebrew/homebrew-core/blob/master/Formula/rinetd.rb)
- [`rmtrash`](https://github.com/Homebrew/homebrew-core/blob/master/Formula/rmtrash.rb)
- [`svdlibc`](https://github.com/Homebrew/homebrew-core/blob/master/Formula/svdlibc.rb)
- [`vip`](https://github.com/Homebrew/homebrew-core/blob/master/Formula/vip.rb)

Reply:
@Rylan12 That's a good point, I can add your change for those cases where the version isn't in the URL.

@jonchang  is right - for the Git tag/revision formulae, you can't have both a `sha256` and a `tag/revision`: https://github.com/Homebrew/brew/blob/master/Library/Homebrew/rubocops/components_redundancy.rb#L31